File: iqchannels-sdk/src/main/java/ru/iqchannels/sdk/schema/ChatMessageForm.java
package ru.iqchannels.sdk.schema;

import androidx.annotation.Nullable;

/**
 * Copyright iqstore.ru. All Rights Reserved.
 */
public class ChatMessageForm {
    public long LocalId;
    public String Payload; // ChatPayloadType
    public String Text;
    @Nullable public String FileId;
    @Nullable public Long ReplyToMessageId;
    @Nullable public String BotpressPayload;

    public ChatMessageForm() {}

    public static ChatMessageForm text(long localId, String text, Long replyToMessageId) {
        ChatMessageForm form = new ChatMessageForm();
        form.LocalId = localId;
        form.Payload = ChatPayloadType.TEXT;
        form.Text = text;
        form.ReplyToMessageId = replyToMessageId;
        return form;
    }

    public static ChatMessageForm file(long localId, String fileId, Long replyToMessageId) {
        ChatMessageForm form = new ChatMessageForm();
        form.LocalId = localId;
        form.Payload = ChatPayloadType.FILE;
        form.FileId = fileId;
        form.ReplyToMessageId = replyToMessageId;
        return form;
    }

    public static ChatMessageForm payloadReply(long localId, String text, String botpressPayload) {
        ChatMessageForm form = new ChatMessageForm();
        form.LocalId = localId;
        form.Payload = ChatPayloadType.TEXT;
        form.Text = text;
        form.BotpressPayload = botpressPayload;
        return form;
    }
}

File: iqchannels-sdk/src/main/java/ru/iqchannels/sdk/schema/ChatMessageForm.kt
package ru.iqchannels.sdk.schema

class ChatMessageForm {
	var LocalId: Long = 0
	var Payload: String? = null // ChatPayloadType
	var Text: String? = null
	var FileId: String? = null
	var ReplyToMessageId: Long? = null
	var BotpressPayload: String? = null
	var ChatType: String = ru.iqchannels.sdk.domain.models.ChatType.REGULAR.name.lowercase()

	companion object {
		fun text(localId: Long, text: String?, replyToMessageId: Long?): ChatMessageForm {
			val form = ChatMessageForm()
			form.LocalId = localId
			form.Payload = ChatPayloadType.TEXT
			form.Text = text
			form.ReplyToMessageId = replyToMessageId
			return form
		}

		fun file(localId: Long, text: String?, fileId: String?, replyToMessageId: Long?): ChatMessageForm {
			val form = ChatMessageForm()
			form.LocalId = localId
			form.Payload = ChatPayloadType.FILE
			form.FileId = fileId
			form.Text = text
			form.ReplyToMessageId = replyToMessageId
			return form
		}

		fun payloadReply(localId: Long, text: String?, botpressPayload: String?): ChatMessageForm {
			val form = ChatMessageForm()
			form.LocalId = localId
			form.Payload = ChatPayloadType.TEXT
			form.Text = text
			form.BotpressPayload = botpressPayload
			return form
		}
	}
}
